core/primitives/impl/scale_codec_impl.cpp

```diff
@@ -84,6 +75,30 @@ namespace kagome::scale {
     }
   };
 
+  /// encodes parachain::Chain
+  template <>
+  struct TypeEncoder<primitives::parachain::Chain> {
+    outcome::result<void> encode(const primitives::parachain::Chain &value,
+                                 common::Buffer &out) {
+      return variant::encodeVariant(value, out);
+    }
+  };
+
+  /// encodes object which has no content
+  template <class T>
+  struct EmptyEncoder {
+    outcome::result<void> encode(const T &value, common::Buffer &out) {
+      return outcome::success();  // no content
+    }
+  };
+
+  /// encodes object of class primitives::parachain::Relay
+  template <>
+  struct TypeEncoder<primitives::parachain::Relay>
+      : public EmptyEncoder<primitives::parachain::Relay> {};
```

Comment:
where is implementation?

Reply:
Implementation is in public EmptyEncoder<primitives::parachain::Relay>;
no additional code is required.
the parachain::Relay itself has no content "for now" as written in polkadot rust.